compiler/crates/relay-typegen/tests/generate_typescript/fixtures/fragment-spread.expected

```diff
@@ -48,137 +48,118 @@ fragment UserFrag2 on User {
   __typename
 }
 ==================================== OUTPUT ===================================
-import type { PageFragment$ref } from "PageFragment.graphql";
-import { FragmentReference } from "relay-runtime";
-export type ConcreateTypes$ref = FragmentReference & { _: "ConcreateTypes$ref" };
-export type ConcreateTypes$fragmentType = ConcreateTypes$ref & { _: "ConcreateTypes$fragmentType" };
+import { FragmentRefs } from "relay-runtime";
 export type ConcreateTypes = {
-  readonly actor?: {
+  readonly actor: {
     readonly __typename: "Page",
     readonly id: string,
-    readonly $fragmentRefs: PageFragment$ref,
+    readonly " $fragmentRefs": FragmentRefs<"PageFragment">,
   } | {
     readonly __typename: "User",
-    readonly name?: string,
+    readonly name: string | null,
   } | {
     // This will never be '%other', but we need some
     // value in case none of the concrete values match.
     readonly __typename: "%other"
-  },
-  readonly $refType: ConcreateTypes$ref,
+  } | null,
+  readonly " $refType": FragmentRefs<"ConcreateTypes">,
 };
 export type ConcreateTypes$data = ConcreateTypes;
 export type ConcreateTypes$key = {
-  readonly $data?: ConcreateTypes$data,
-  readonly $fragmentRefs: ConcreateTypes$ref,
+  readonly " $data"?: ConcreateTypes$data,
+  readonly " $fragmentRefs": FragmentRefs<"ConcreateTypes">,
 };
 -------------------------------------------------------------------------------
-import type { OtherFragment$ref } from "OtherFragment.graphql";
-import type { PictureFragment$ref } from "PictureFragment.graphql";
-import type { UserFrag1$ref } from "UserFrag1.graphql";
-import type { UserFrag2$ref } from "UserFrag2.graphql";
-import { FragmentReference } from "relay-runtime";
-export type FragmentSpread$ref = FragmentReference & { _: "FragmentSpread$ref" };
-export type FragmentSpread$fragmentType = FragmentSpread$ref & { _: "FragmentSpread$fragmentType" };
+import { FragmentRefs } from "relay-runtime";
 export type FragmentSpread = {
   readonly id: string,
-  readonly justFrag?: {
-    readonly $fragmentRefs: PictureFragment$ref
-  },
-  readonly fragAndField?: {
-    readonly uri?: string,
-    readonly $fragmentRefs: PictureFragment$ref,
-  },
-  readonly $fragmentRefs: OtherFragment$ref & UserFrag1$ref & UserFrag2$ref,
-  readonly $refType: FragmentSpread$ref,
+  readonly justFrag: {
+    readonly " $fragmentRefs": FragmentRefs<"PictureFragment">
+  } | null,
+  readonly fragAndField: {
+    readonly uri: string | null,
+    readonly " $fragmentRefs": FragmentRefs<"PictureFragment">,
+  } | null,
+  readonly " $fragmentRefs": FragmentRefs<"OtherFragment" | "UserFrag1" | "UserFrag2">,
+  readonly " $refType": FragmentRefs<"FragmentSpread">,
```

Comment:
@kassens Just a FYI: If you were to switch the Flow emission to use types like these (based on strings) as well, instead of opaque types, we could get rid of any additional code related to haste or a specific artifact directory.

Reply:
Oh, interesting. This is using the fragment names and the fact that they're globally unique as their identifier, clever!

What about the property names? Why `" $refType"` as the key (leading space?)

Reply:
So I wasn't involved with the original design behind the Typescript types, but I believe the leading space causes Typescript to omit this property in its autocomplete.

Reply:
> So I wasn't involved with the original design behind the Typescript types, but I believe the leading space causes Typescript to omit this property in its autocomplete.

Correct 👍